pkg/apis/flyteworkflow/v1alpha1/node_status.go

```diff
@@ -180,6 +180,8 @@ type NodeStatus struct {
 
 	TaskNodeStatus    *TaskNodeStatus    `json:",omitempty"`
 	DynamicNodeStatus *DynamicNodeStatus `json:"dynamicNodeStatus,omitempty"`
+	// In case of Failing/Failed Phase, an execution error can be optionally associated with the Node
+	Err *ExecutionError `json:"error,omitempty"`
```

Comment:
Nit: I would match the name in json and the field... 

```suggestion
	Error *ExecutionError `json:"error,omitempty"`
```

Reply:
I guess that should work, lower case `error` is a reserved keyword